src/select_all.rs

```diff
@@ -72,14 +72,14 @@ impl<A> Future for SelectAll<A>
     }
 
     fn schedule(&mut self, task: &mut Task) {
-        for f in self.inner.iter_mut() {
+        for f in &mut self.inner {
```

Comment:
I personally prefer `.iter_mut()` instead of `&mut foo`, could this stay the same?